### DISCUSSION TITLE
Script for changing the layer exposure time of the internal layers of a model ?

### DISCUSSION THREAD

**Discussion by susheemkanwar** [sn4k3/UVtools#942] [Scripts]
Sorry I don't have a coding background and am completely new to this. What is the procedure to accomplish something like this? Do you write the code in let's say VS code using Python and import it into UVTools? How does the scripting work exactly? 

I specifically want a script that can help me change the layer exposure time of the 3D model to a lower one. How do I go about accomplishing that?

**Comment by sn4k3** [comment]
You can start here: https://github.com/sn4k3/UVtools/tree/master/Scripts/UVtools.ScriptSample
Read the page and explore the samples. 
But if you only want to change exposure time of the model you can use the GUI for that.
For the script the command is: `SlicerFile.BottomExposureTime = 30f;` and `SlicerFile.ExposureTime = 3f;`

**Reply by susheemkanwar** [reply]
Thank you, no I specifically want to use the script and not the GUI because later I will modify the other variables. Thank you for sharing the link. I will look at the samples. Greatly appreciate it.

**Comment by susheemkanwar** [comment]
So, I tried the script samples available in documentation and am able to manipulate the variables I want, let's say the layer exposure time. However, is there a way to bypass using the UVTools GUI and directly execute the code from the VS Code terminal. 

**For example,** let's say I have the output of a sliced file in .ctb format that has 10 layers of 1s exposure time each. My goal is to change the exposure time to 5s each. One way is to use a script that changes that Layer.ExposureTime and import that C# script into UVTools using Tools->Scripting->Load Script. 

**However,** if I don't want to load the C# script using UVTools but input it directly can I do that? What I mean is can I input my .ctb file into my C# script somehow, then change the layer exposure time and finally save that output. So that now if I input this .ctb file into UVTools it would show me the layer exposure time as 5s.

Thank you for all your help.

**Reply by sn4k3** [reply]
You can use the [UVtoolsCmd](https://github.com/sn4k3/UVtools?tab=readme-ov-file#uvtoolscmd-console-executable) for that. eg:  `UVtoolsCmd.exe run <input-file> <script file(s)>`

**Reply by susheemkanwar** [reply]
When I try and do that it throws an error saying syntax incorrect (attached below)
![Screenshot 2024-12-06 160904](https://github.com/user-attachments/assets/4254a9b5-af08-486c-ada7-1e72eaa8f2e1)
.

**Reply by sn4k3** [reply]
Do not use `<` and `>` that represent an required argument, while `[` and `]` represent optional arguments.
Correct syntax is: `UVtoolsCmd run "yourfile.ctb" "yourscript.cs"`

**Reply by susheemkanwar** [reply]
Thank you this worked.
